### PR TITLE
Performance v2

### PR DESCRIPTION
**Pull request by arikalon1**

### CODE REVIEW

File: src/robusta/patch/patch.py [hunk 1]
```diff
@@ -0,0 +1,122 @@
+from typing import Union, List, Dict, get_type_hints
+from dataclasses import is_dataclass, InitVar
+from inspect import signature, getmodule
+from hikaru import HikaruDocumentBase, HikaruBase
+from ruamel.yaml import YAML
+
+try:
+    from typing import get_args, get_origin
+except ImportError:  # pragma: no cover
+    def get_args(tp):
```
Comment (aantn) [inline src/robusta/patch/patch.py:10]: This is weird. I'm sure there is a good reason for it though! Perhaps add a comment explaining why we're doing this?

File: src/robusta/patch/patch.py [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+        return tp.__args__ if hasattr(tp, "__args__") else ()
+
+    def get_origin(tp):
+        return tp.__origin__ if hasattr(tp, "__origin__") else None
+
+NoneType = type(None)
+
+
+def create_monkey_patches():
+    print("Creating hikaru monkey patches")
```
Comment (aantn) [inline src/robusta/patch/patch.py:20]: Use `logging` not `print`

File: src/robusta/patch/patch.py [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+    print("Creating yaml monkey patch")
+    YAML.official_plug_ins = official_plug_ins
+
+
+def official_plug_ins(self):
+    return []
+
+# hikaru meta.py monkey patch function
+@classmethod
+def get_empty_instance(cls):
```
Comment (aantn) [inline src/robusta/patch/patch.py:32]: I would explicitly add a comment marking the areas that are your changes vs the orignal

File: src/robusta/patch/patch.py [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+    def get_args(tp):
+        return tp.__args__ if hasattr(tp, "__args__") else ()
+
+    def get_origin(tp):
+        return tp.__origin__ if hasattr(tp, "__origin__") else None
+
+NoneType = type(None)
+
+
+def create_monkey_patches():
```
Comment (aantn) [inline src/robusta/patch/patch.py:19]: I would add a comment here explaining the motivation for each of these patches.

**Review by aantn — COMMENTED** [review]
Didn't look at the actual patches in depth, but looks good.
Left some minor comments mostly regarding comments.

File: src/robusta/patch/patch.py (hunk 1 above)
Comment (arikalon1) [reply src/robusta/patch/patch.py:10]: I'm not sure tbh. I just copied that section from Hikaru meta.py
Do you think we should change something here?

File: src/robusta/patch/patch.py (hunk 2 above)
Comment (arikalon1) [reply src/robusta/patch/patch.py:20]: Correct, fixing

File: src/robusta/patch/patch.py (hunk 3 above)
Comment (arikalon1) [reply src/robusta/patch/patch.py:32]: adding it

File: src/robusta/patch/patch.py (hunk 4 above)
Comment (arikalon1) [reply src/robusta/patch/patch.py:19]: Adding

File: src/robusta/patch/patch.py (hunk 1 above)
Comment (aantn) [reply src/robusta/patch/patch.py:10]: No it's fine. Probably just to support older python versions

**Pull request merged**